client/src/modules/journal/journal.js

```diff
@@ -277,6 +277,33 @@ function JournalController(Journal, Sorting, Grouping,
   // API register function
   function onRegisterApi(gridApi) {
     vm.gridApi = gridApi;
+    
+    vm.gridApi.edit.on.afterCellEdit($scope, function(rowEntity, colDef, newValue, oldValue){
+      if(newValue != oldValue) {
+        propagate(colDef.field,newValue);
+      }
+    });
+  }
+
+  function propagate(column, value){
+    var propagateColumn = ['trans_date', 'entity_uuid', 'origin_id'];
+    
+    //Check if the column updated must be propragated in all transaction
+    var checkPropagate = (propagateColumn.indexOf(column));
+
+    if(checkPropagate !== -1){
+      // For the old row
+      vm.transactions._entity.data.data.forEach(function (row) {
```

Comment:
Lines 296-299 is the same code as lines 302-305.  Let's make it into the same funtion:

```js
function updateSharedPropertyOnRow(row) {
  transactions.editCell(row, column, value, row[column]);
  row[column] = (column === 'trans_date') ? new Date(value) : value;
}

// pass updates on to both the original rows and the new (pending) rows
vm.transactions._entity.data.data.forEach(updateSharedPropertyOnRow);
vm.transactions._entity.newRows.data.forEach(updateSharedPropertyOnRow);
```